pipeline/processor/postprocessor.py

```diff
@@ -196,6 +198,67 @@ def merge_text_blocks(self, text_blocks: List[dict]) -> dict:
         }
         return text_block
 
+    @staticmethod
+    def _get_all_paths_to_merge(
+        all_paths: List[tuple], path: tuple
+    ) -> Tuple[List[int], List[int]]:
+        """Get all paths on a page that are inferred to be part of the same semantics. See below for detail.
+
+        Some blocks should be merged with other paths (are part of the same semantics) even though their
+        own paths aren't duplicated. This is because Adobe's style detector fails to detect all
+        styled elements with metadata tags, in which case styled elements are contained in separate blocks with
+        different paths. For example, three contiguous blocks with paths P/L/, P/L/Span and P/L should all be joined
+        but relying on Adobe tags and duplicated path names alone will ignore the middle element.
+        This is always the case for italics, which Adobe never detects (especially problematic as we'd expect
+        italicisation to be important a-priori (it indicates emphasis or references to other documents). It also often
+        happens for bolded and underlined elements for an unknown reason.
+
+        Args:
+            all_paths: The path of every block on the page.
+            path: The duplicated path.
+
+        Returns:
+            A list of all paths to merge and a list of the subset of these paths that are duplicates in Adobe output.
+        """
+        duplicated_path_indices = []
+        for i, j in enumerate(all_paths):
+            if path == j:
+                duplicated_path_indices.append(i)
+        merge_path_indices = list(
+            range(duplicated_path_indices[0], duplicated_path_indices[-1] + 1)
+        )
+        return merge_path_indices, duplicated_path_indices
+
+    @staticmethod
+    def _update_style_metadata(
```

Comment:
This function is not strictly necessary. It's just a helper util for grabbing metadata on the path type of the newly included semantics. Unfortunately, this won't be "Bold", "Underline" etc, but is more likely to be "span", which may be somewhat helpful for visibility and debugging down the line.